docs/source/agent.rst

```diff
@@ -25,33 +25,35 @@ In order to satisfy above requirements, we opted for a simple thread-based appro
 all software components that could potentially block into their own threads. In particular, these
 are ``ContractEventMonitor`` and ``EventProcessor``.
 
+With the current implementation, we have a ``EventProcessor`` for every direction in their own thread running for the agent.
+This helps to scale well, as if there is a new L2 chain added, then agent would just create another directions which has the new
+chain and there is no need to run seperate processes for this manner. Also, we have L2 chain based ``ContractEventMonitors`` that 
+event processors can subscribe and get events from. This prevents us to create redundant event monitors for event processors.
+
```

Comment:
I don't think we need to mention the older, multiple-agent-processes implementation here, it's just something that we had and is now gone.

Please also don't make the lines in paragraph too long and try to follow the rest of the document.